### ISSUE TITLE
EPIC: `Marketplace` - Improve the checkout experience

### ISSUE THREAD

**Issue by KellyAH** [zinc-collective/convene#1672]
- improvements to https://github.com/zinc-collective/convene/issues/831

This is an attempt to focus our efforts on improving/fixing the currently most valuable area of the app, the `Marketplace` which is used by [Piikup](https://github.com/zinc-collective/convene/issues/832) and other clients.

# Tasks 
## 🐛 Fix bugs
Anyone can pick up fixing a bug 😄 
- [x] https://github.com/zinc-collective/convene/issues/1657
- [x] https://github.com/zinc-collective/convene/issues/1656
- [x] https://github.com/zinc-collective/convene/issues/1658
- [x] https://github.com/zinc-collective/convene/issues/1660

## Feature

**Marketplace buyers can see when their order will arrive**
See - [Project doc](https://docs.google.com/document/d/1wbXO82OeEkoIOj8zL6B24fiCC2Wg-w7949fo6wg3qhc/edit?usp=sharing)
- [x] MVP  - @daltonrpruitt and @KellyAH will pair on 7/25 https://github.com/zinc-collective/convene/pull/1705
- [x] https://github.com/zinc-collective/convene/issues/1509
- [ ] V1 - **not doing right now - out of scope!**

**Comment by anaulin** [comment]
Catching up with this project plan today (it's been that kind of week...). Thanks for putting this together, @KellyAH, I found it very helpful to see it all written up and also the screenshots with examples of how other apps do it. 😻

I wanted to offer some thoughts on the MVP, in case these are helpful for the implementation:
> Always Display marketplace delivery window value on the marketplace page. Either at the top or bottom where total price is displayed. 

In our current architecture, the delivery window depends on the delivery area chosen by the shopper. So we don't know the delivery window until the shopper chooses a delivery area -- that's why at the moment we don't show it until the "delivery details" are entered by the shopper.

The way I see other delivery sites dealing with this is that either they just have one delivery area for the entire marketplace, or they ask you upfront before you start shopping to enter a zip code or address or otherwise choose a delivery area. We could do something similar in Convene:
* if Marketplace has only one delivery area
  * all orders will default to this delivery are, we don't need shoppers to pick one 
  * we can show the delivery windows and delivery information on the Marketplace without any need for shopper input
* if Marketplace has more than one delivery area
  * we will need the shopper to pick a delivery area before we can show them the delivery windows
  * we can't show the delivery window information upfront, but instead we could show some text that says "please select a delivery area to see delivery window information" in that same spot, with a link to select the delivery area

These are just some rough thoughts based on my mental model of the marketplace, I'm sure other (better?) approaches are possible.

**Comment by KellyAH** [comment]
Glad to see the project plan is helpful. 

I am very happy and willing to write up more project plans for convene whenever it is needed/wanted to help add clarity and scope, etc. As a technical writer, I enjoy writing them and I also gain a lot of clarity and big-picture scope from them. 

And right now I find writing up project plans more enjoyable than app coding.

I would just need others to review the project plans for accuracy.

**Comment by zspencer** [comment]
I've really appreciated having the set of bugs that are most important laid out; as well as knowing that y'all are taking responsibility for polishing up the Delivery UX!

**Comment by zspencer** [comment]
@KellyAH, if you get a chance can you run through the checkout experience and let me know if you think it meets the intent of the epic?

**Comment by KellyAH** [comment]
✅ confirmed the checkout experience is better! Customers now can more easily tell when orders will arrive based on the delivery location and order time.
